test/compile/data/bin.test.ts

```diff
@@ -50,18 +50,60 @@ describe('compile/data/bin', function() {
         assert.deepEqual(transform[0], {
           type: 'extent',
           field: 'Acceleration',
-          signal: 'Acceleration_extent'
+          signal: "bin_maxbins_10_Acceleration_extent"
         });
         assert.deepEqual(transform[1], {
           type: 'bin',
           field: 'Acceleration',
-          as: ['bin_Acceleration_start', 'bin_Acceleration_end'],
+          as: ["bin_maxbins_10_Acceleration_start", "bin_maxbins_10_Acceleration_end"],
           maxbins: 10,
-          signal: "Acceleration_bins",
-          extent: {signal: 'Acceleration_extent'}
+          signal: "bin_maxbins_10_Acceleration_bins",
+          extent: {signal: 'bin_maxbins_10_Acceleration_extent'}
         });
       });
     });
+    describe('should bin the same field', () => {
+      const model = parseUnitModel({
+        data: {"url": "data/movies.json"},
+        mark: "circle",
+        encoding: {
+          x: {
+            bin: {maxbins: 10},
+            field: "IMDB_Rating",
+            type: "q"
+          },
+          y: {
+            bin: {"maxbins": 10},
+            field: "Rotten_Tomatoes_Rating",
+            type: "q"
+          },
+          color: {
+            bin: {maxbins: 20},
+            field: "Rotten_Tomatoes_Rating",
+            type: "quantitative"
+          },
+          size: {
+            aggregate: "count",
+            field: "*",
+            type: "q"
+          }
+        }
+      });
+      const transform = vals(bin.parseUnit(model))[0];
```

Comment:
You test here doesn't even check the field that is used twice!  

Make sure to (1) check that and (2) check that the output bin component only has two items (not three!)